app/models/track.rb

```diff
@@ -36,7 +36,7 @@ class Track < ApplicationRecord
   scope :by_filter, ->(filter) { where('"tracks"."normalized_title" LIKE ?', "%#{Track.normalize(filter)}%") }
   scope :by_artist, ->(artist) { joins(:artists).where(artists: { id: artist }) }
   scope :by_album, ->(album) { where(album:) }
-  scope :by_genre, ->(genre) { joins(:genres).where(genres: { id: genre }) }
+  scope :by_genre, ->(genre) { where(id: joins(:genres).where(genres: { id: genre }).select(:id)) }
```

Comment:
I think it should be a bit more efficient to do this:
```suggestion
  scope :by_genre, ->(genre) { where(id: TrackGenre.where(genre_id: genre).select(:track_id)) }
```
Right now the query that was built up before gets copied to the subquery, which means all conditions would have to be executed twice. It doesn't really matter for our usecases though since AFAIK we only ever filter by one scope in existing frontends.

Reply:
since we use `has_and_belongs_to_many`, we don't actually have a `TrackGenre` model.
I though the implicit self before `joins(:genres)` refered to the `Track` class and not the query chain. Maybe we just need to explicitly mention the class to escape the query that is being build up?
```
scope :by_genre, ->(genre) { where(id: Track.joins(:genres).where(genres: { id: genre }).select(:id)) }
```

Reply:
That would work. (I didn't realize that we used `has_and_belongs_to_many`.)